pkg/ocmagenthandler/ocmagenthandler_networkpolicy.go

```diff
@@ -0,0 +1,110 @@
+package ocmagenthandler
+
+import (
+	"context"
+	"reflect"
+
+	netv1 "k8s.io/api/networking/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	ocmagentv1alpha1 "github.com/openshift/ocm-agent-operator/pkg/apis/ocmagent/v1alpha1"
+	oah "github.com/openshift/ocm-agent-operator/pkg/consts/ocmagenthandler"
+)
+
+func buildNetworkPolicy() netv1.NetworkPolicy {
+	namespacedName := oah.BuildNamespacedName(oah.OCMAgentNetworkPolicyName)
+	np := netv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      namespacedName.Name,
+			Namespace: namespacedName.Namespace,
+		},
+		Spec: netv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": oah.OCMAgentName,
```

Comment:
do we want to restrict the access to the ocm-agent pods only or the whole project?